unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs

```diff
@@ -109,29 +118,131 @@ public void SetVisibility(bool visible)
             HideMentionSuggestions();
     }
 
-    public void AddChatMessage(ChatMessage message, bool setScrollPositionToBottom = false, bool spamFiltering = true, bool limitMaxEntries = true)
+    public void AddChatMessage(ChatMessage message, bool setScrollPositionToBottom = false,
+        bool spamFiltering = true, bool limitMaxEntries = true, int fetchProfileTries = 0)
     {
-        AddChatMessage(ChatMessageToChatEntry(message), setScrollPositionToBottom, spamFiltering, limitMaxEntries).Forget();
+        async UniTaskVoid EnsureProfileThenUpdateMessage(string profileId, ChatMessage message,
+            bool setScrollPositionToBottom, bool spamFiltering, bool limitMaxEntries,
+            int fetchProfileTries,
+            CancellationToken cancellationToken)
+        {
+            if (fetchProfileTries >= MAX_FETCH_PROFILE_TRIES)
+            {
+                Debug.LogWarning($"Profile fetching max retries limit reached for {profileId}");
+                return;
+            }
+
+            try
+            {
+                UserProfile requestedProfile = await userProfileBridge.RequestFullUserProfileAsync(profileId, cancellationToken);
+
+                if (requestedProfile != null)
```

Comment:
Just a highlight: according to the logic underneath `userProfile` can never be `null`